_aulas/15/06_extra_scraping_parsing.md

```diff
@@ -0,0 +1,44 @@
+---
+layout: conteudo_geral
+modulo: 15
+numero: 6
+nome: "Scraping - Parsing"
+title: "15.6 [Extra] Scraping | Estágio em Programação"
+description: "Aprenda sobre Scraping do curso online
+  gratuito Estágio em Programação."
+---
+
+A tarefa de processar um HTML em busca de certos conteúdos é difícil de se executar utilizando apenas funções básicas de processamento de texto como buscas de palavras, substituições ou loops de leitura de caracteres.
+
+Duas ferramentas são muito importantes no parsing de documentos HTML: **seletores CSS** e **expressões regulares**.
+
+### Seletor CSS
+
+Para fazer parsing de HTML utilizando seletores CSS, utiliza-se um parser para Ruby chamado Nokogiri. Sua instalação pode ser realizada com o seguinte comando:
+
+```
+  gem install nokogiri
+```
+
+Veja no exemplo abaixo como seria o uso de seletores CSS usando o Nokogiri:
+
+```ruby
+  require 'net/http'
+  require 'nokogiri'
+
+  # Download a product URL
+  url = 'http://www.goldentoys.com.br/funko-pop-knightmare-batman'
+  body = Net::HTTP.get(URI.parse(url))
+
+  # Parse the body html with Nokogiri
+  doc = Nokogiri::HTML(body)
+
+  # Print the price of the product using CSS selector
+  puts doc.css('meta[itemprop="price"]').first.attr(:content)
+```
+
+### Expressão regular
+
+Expressões regulares são úteis para extrair conteúdos que seguem um certo padrão em um texto.
+
+Ferramenta online para testar expressão regular em Ruby: [rubular](https://rubular.com/).
```

Comment:
Colocar link interno para capítulo de regex no estágio em programação.

Reply:
Acho que se for testar regex é melhor ser um link somente para isso, ou então, faço um comentário no texto sobre o módulo de regex e coloco o link interno.